pkg/agent/controller/networkpolicy/reject.go

```diff
@@ -115,8 +116,26 @@ func (c *Controller) rejectRequest(pktIn *ofctrl.PacketIn) error {
 		isIPv6 = true
 	}
 
-	sIface, srcFound := c.ifaceStore.GetInterfaceByIP(srcIP)
-	dIface, dstFound := c.ifaceStore.GetInterfaceByIP(dstIP)
+	sIface, srcIsLocal := c.ifaceStore.GetInterfaceByIP(srcIP)
+	dIface, dstIsLocal := c.ifaceStore.GetInterfaceByIP(dstIP)
+
+	isFlexibleIPAMSrc, isFlexibleIPAMDst, ctZone, err := parseFlexibleIPAMStatus(pktIn, c.nodeConfig, srcIP, srcIsLocal, dstIP, dstIsLocal)
+	if err != nil {
+		return err
+	}
+	matches := pktIn.GetMatches()
+	if (isFlexibleIPAMSrc || isFlexibleIPAMDst) && (srcIsLocal && dstIsLocal) {
+		// For an ingress reject rule with AntreaIPAM cross-VLAN traffic, the original packet comes from uplink/gw, and
```

Comment:
"the uplink / gateway port" to be clearer.

Reply:
Fixed.